tests/lib/rules/no-misleading-unicode-character.ts

```diff
@@ -262,5 +269,52 @@ tester.run("no-misleading-unicode-character", rule as any, {
             options: [{ fixable: true }],
             errors: [{ messageId: "characterClass" }],
         },
+
+        // ES2024
+        {
+            code: String.raw`/[[👶🏻]]/v`,
+            output: String.raw`/[[\q{👶🏻}]]/v`,
+            options: [{ fixable: true }],
+            errors: [{ messageId: "characterClass" }],
+        },
+        {
+            code: String.raw`/[👶🏻[👨‍👩‍👦]]/v`,
+            output: String.raw`/[\q{👶🏻}[👨‍👩‍👦]]/v`,
+            options: [{ fixable: true }],
+            errors: [
+                { messageId: "characterClass", column: 3 },
+                { messageId: "characterClass", column: 8 },
+            ],
+        },
+        {
+            code: String.raw`/[👶🏻👨‍👩‍👦]/v`,
+            output: String.raw`/[\q{👶🏻|👨‍👩‍👦}]/v`,
+            options: [{ fixable: true }],
+            errors: [{ messageId: "characterClass" }],
+        },
+        {
+            code: String.raw`/[👶🏻&👨‍👩‍👦]/v`,
+            output: String.raw`/[\q{👶🏻|👨‍👩‍👦}&]/v`,
```

Comment:
Could this potentially create invalid regexes? `\q` isn't allowed in negated character classes, so e.g. `[^👶🏻&👨‍👩‍👦]` can't be fixed to `[^\q{👶🏻|👨‍👩‍👦}&]`. Does this work correctly? If yes, could we add a test for this?

Reply:
Thank you for checking it!
I added a test case. It has already been avoided in L137 and seems to be fine.